hikari/events/member_events.py

```diff
@@ -98,6 +98,9 @@ def guild(self) -> typing.Optional[guilds.GatewayGuild]:
             The guild that this event occurred in, if known, else
             `builtins.None`.
         """
+        if not isinstance(self.app, traits.CacheAware):
+            return None
```

Comment:
why not just 
```py
if isinstance(self.app, traits.CacheAware):
    return elf.app.cache.get_available_guild(self.guild_id) or self.app.cache.get_unavailable_guild(self.guild_id)
```
and let it implicitly return None otherwise for these 1 liners?

Reply:
Mypy doesnt allow functions that say that return a union of something and `None` to return `None` in an indirect way.